src/app.rs

```diff
@@ -392,6 +418,24 @@ impl<T: Send + 'static, V: View<T> + 'static> App<T, V> {
         disable_raw_mode()?;
         Ok(())
     }
+
+    #[cfg(test)]
+    pub fn get_event_tx(&self) -> tokio::sync::mpsc::Sender<Event> {
+        self.event_tx.clone()
+    }
+
+    #[cfg(test)]
+    pub fn get_terminal(&mut self) -> &mut Terminal<TestBackend> {
+        &mut self.terminal
+    }
+}
+
+/// Restore the terminal no matter how the app exits
+impl<T: Send + 'static, V: View<T> + 'static> Drop for App<T, V> {
+    fn drop(&mut self) {
+        self.cleanup()
+            .unwrap_or_else(|e| eprint!("Restoring the terminal failed: {e}"));
+    }
```

Comment:
One thing to keep in mind:

Currently the real terminal/stdout is mutated in tests when I see it right (`EnableFocusChange`, `EnableMouseCapture`...). It would be good to redirect or disable these mutations.


Hmm well thinking about it, a `#[cfg(not(test))]` above each of the `execute!(stdout(),..)` should be enough already, so maybe add this as well. And I think above `enable/disable_raw_mode` as well.

Reply:
This shouldn't be necessary. Tests use `Terminal<TestBackend>` for widget and view tests.

| /// A [`Backend`] implementation used for integration testing that that renders to an in memory
/// buffer.

Reply:
Ok, I fixed it.